tesseract_common/include/tesseract_common/class_loader.h

```diff
@@ -90,10 +112,31 @@ struct ClassLoader
 };
 }  // namespace tesseract_common
 
-#define TESSERACT_ADD_PLUGIN(DERIVED_CLASS, ALIAS)                                                                     \
+// clang-format off
+#define TESSERACT_ADD_PLUGIN_SECTIONED(DERIVED_CLASS, ALIAS, SECTION)                                                  \
   extern "C" BOOST_SYMBOL_EXPORT DERIVED_CLASS ALIAS;                                                                  \
+  BOOST_DLL_SECTION(SECTION, read) BOOST_DLL_SELECTANY                                                                 \
   DERIVED_CLASS ALIAS;
 
+#define TESSERACT_ADD_PLUGIN(DERIVED_CLASS, ALIAS)                                                                     \
+  TESSERACT_ADD_PLUGIN_SECTIONED(DERIVED_CLASS, ALIAS, boostdll)
+
+#define TESSERACT_ADD_FWD_KIN_PLUGIN(DERIVED_CLASS, ALIAS)                                                             \
+  TESSERACT_ADD_PLUGIN_SECTIONED(DERIVED_CLASS, ALIAS, FwdKin)
+
+#define TESSERACT_ADD_INV_KIN_PLUGIN(DERIVED_CLASS, ALIAS)                                                             \
+  TESSERACT_ADD_PLUGIN_SECTIONED(DERIVED_CLASS, ALIAS, InvKin)
+
+#define TESSERACT_ADD_DISCRETE_MANAGER_PLUGIN(DERIVED_CLASS, ALIAS)                                                    \
+  TESSERACT_ADD_PLUGIN_SECTIONED(DERIVED_CLASS, ALIAS, DiscColl)
+
+#define TESSERACT_ADD_CONTINUOUS_MANAGER_PLUGIN(DERIVED_CLASS, ALIAS)                                                  \
+  TESSERACT_ADD_PLUGIN_SECTIONED(DERIVED_CLASS, ALIAS, ContColl)
+
+#define TESSERACT_ADD_VISUALIZATION_PLUGIN(DERIVED_CLASS, ALIAS)                                                       \
+  TESSERACT_ADD_PLUGIN_SECTIONED(DERIVED_CLASS, ALIAS, Plotter)
```

Comment:
It might make more sense to define these macros in the respective packages where their plugin interface is defined. If I were making a new plugin, my first thought would be to look in the package relating to that plugin type to get this macro.

Reply:
Done